libminifi/test/unit/ResourceQueueTests.cpp

```diff
@@ -76,20 +81,39 @@ TEST_CASE("Resource limitation is not set to the resource queue", "[utils::Resou
 }
 
 TEST_CASE("resource returns when it goes out of scope", "[utils::ResourceQueue]") {
-  auto queue = utils::ResourceQueue<int>::create(std::nullopt, nullptr);
+  using std::chrono::steady_clock;
+  auto queue = utils::ResourceQueue<steady_clock::time_point>::create([]{ return std::make_unique<steady_clock::time_point>(steady_clock::time_point::min()); });
+  {
+    auto resource = queue->getResource();
+    CHECK(*resource == steady_clock::time_point::min());
+    *resource = steady_clock::now();
+  }
+  {
+    auto resource = queue->getResource();
+    CHECK(*resource != steady_clock::time_point::min());
+  }
+}
+
+TEST_CASE("resource resets when it goes out of scope", "[utils::ResourceQueue]") {
+  using std::chrono::steady_clock;
+  auto queue = utils::ResourceQueue<steady_clock::time_point>::create([]{ return std::make_unique<steady_clock::time_point>(steady_clock::time_point::min()); },
+                                                                      std::nullopt,
+                                                                      [](steady_clock::time_point& resource){ resource = steady_clock::time_point::max();});
```

Comment:
A better way to test this would be to have the reset function assign the same initial value as the create function, but change the value in the first scope, and check that it's back to the default in the second.
This way you're testing the requirements instead of the implementation.

Reply:
Good idea, I've changed this and also included some additional checks aswell in https://github.com/apache/nifi-minifi-cpp/pull/1473/commits/5279824fbde23389279cce46b2859f979d93044f